allennlp/data/fields/text_field.py

```diff
@@ -67,7 +68,8 @@ def get_padding_lengths(self) -> Dict[str, int]:
             lengths.append(indexer_lengths)
         padding_lengths = {'num_tokens': len(self._indexed_tokens[0])}
         # Get all the keys which have been used for padding.
-        padding_keys = set().union(*[d.keys() for d in lengths])
+        # TODO: figure out the right type hint for this next line
+        padding_keys = set().union(*[d.keys() for d in lengths])  # type: ignore
```

Comment:
it turns it's not, but only because it's written in a roundabout way. I changed it to

```
{key for d in lengths for key in d.keys()}
```

and it checks